silkworm/silkrpc/commands/ots_api.cpp

```diff
@@ -249,6 +251,115 @@ boost::asio::awaitable<void> OtsRpcApi::handle_ots_getBlockTransactions(const nl
+        roaring::Roaring64Map bitmap;
+
+        while (true) {
+            if (key_value.key.empty() || !key_value.key.starts_with(sender_byte_view)) {
+                auto plane_state_cursor = co_await tx->cursor(db::table::kPlainStateName);
+                auto account_payload = co_await plane_state_cursor->seek(sender_byte_view);
+                auto account = Account::from_encoded_storage(account_payload.value).value();
+
+                if (account.nonce > nonce) {
+                    break;
+                }
+
+                reply = make_json_content(request["id"], nullptr);
+                co_await tx->close();
+                co_return;
+            }
+
+            bitmap = silkworm::db::bitmap::parse(key_value.value);
+            auto max_block = bitmap.maximum();
+            auto block_key{silkworm::db::block_key(max_block)};
+            auto account_payload = co_await account_change_set_cursor->seek_both(block_key, sender_byte_view);
+            account_payload = account_payload.substr(sender_byte_view.length());
+            auto account = Account::from_encoded_storage(account_payload).value();
+
+            if (account.nonce > nonce) {
+                break;
+            }
+
+            max_block_prev_chunk = max_block;
+            key_value = co_await account_history_cursor->next();
+        }
+
+        uint64_t cardinality = bitmap.cardinality();
+        blocks = new uint64_t[cardinality];
+        bitmap.toUint64Array(blocks);
+
+        uint64_t idx = 0;
+        for (uint64_t i = 0; i < cardinality; i++) {
+            auto bn = blocks[i];
+            auto bk{silkworm::db::block_key(bn)};
+            auto ap = co_await account_change_set_cursor->seek_both(bk, sender_byte_view);
+            ap = ap.substr(sender_byte_view.length());
+            auto a = Account::from_encoded_storage(ap).value();
+
+            if (a.nonce > nonce) {
+                idx = i;
+                break;
+            }
+        }
+
+        auto nonceBlock = max_block_prev_chunk;
+        if (idx > 0) {
+            nonceBlock = blocks[idx - 1];
+        }
+
+        auto block_with_hash = co_await core::read_block_by_number(*block_cache_, tx_database, nonceBlock);
+
+        for (const auto& item : block_with_hash.block.transactions) {
+            if (item.from == sender && item.nonce == nonce) {
+                auto transaction_hash{hash_of_transaction(item)};
+                auto result = silkworm::to_bytes32({transaction_hash.bytes, silkworm::kHashLength});
+                reply = make_json_content(request["id"], result);
+                co_await tx->close();
+                co_return;
+            }
+        }
+        reply = make_json_content(request["id"], nullptr);
+
+    } catch (const std::invalid_argument& iv) {
+        SILKRPC_WARN << "invalid_argument: " << iv.what() << " processing request: " << request.dump() << "\n";
+        reply = make_json_content(request["id"], nlohmann::detail::value_t::null);
```

Comment:
Why using `nlohmann::detail::value_t::null` and `nullptr` elsewhere? What's the difference?

Please add a comment if there is a good reason for this difference